td/tasks.py

```diff
@@ -1,32 +1,68 @@
 from __future__ import absolute_import
 
-from django.db import connection
-
 from celery import task
+from django.db import connection
 from pinax.eventlog.models import log
 
 from td.imports.models import (
-    EthnologueLanguageCode,
-    EthnologueCountryCode,
-    SIL_ISO_639_3,
-    WikipediaISOLanguage,
-    WikipediaISOCountry,
-    IMBPeopleGroup
-)
-
+    EthnologueLanguageCode, EthnologueCountryCode, SIL_ISO_639_3,
+    WikipediaISOLanguage, WikipediaISOCountry, IMBPeopleGroup)
+from td.models import Region, Country, Language, AdditionalLanguage
 from td.resources.models import Title, Resource, Media
-from td.models import Region, Country, Language
+from td.signals import languages_integrated
+from td.utils import return_ascii_str
+
+
+def _get_anglicized_language_name(row):
+    """
+    Determine which field's value can be used as the `language.anglicized_name`
+    from the language imports integration.
 
-from .models import AdditionalLanguage
-from .signals import languages_integrated
+    :param row: Single row of data from the `intergrate_imports` sql query
+    :type row: tuple
+
+    :returns: str
+    """
+    anglicized_name = ""
+    # If the `native_name` is already an anglicized, return `None`
+    if not return_ascii_str(row[1]):
```

Comment:
I wonder if we should use a `dict` instead of the `list` from `fetchall`, by adding a function like in [this answer](http://stackoverflow.com/a/14294314)